.github/workflows/build_wheels.yml

```diff
@@ -0,0 +1,232 @@
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.platform.target }}
+          args: --release --out dist --interpreter ${{ matrix.platform.interpreter }}
+          sccache: ${{ !startsWith(github.ref, 'refs/tags/') }}
+          manylinux: ${{ matrix.platform.manylinux }}
+
+      - name: Validate Python package distributions
+        run: twine check --strict dist/*
+
+      - name: Install built wheel
+        if: matrix.platform.target == 'x86_64'
+        run: |
+          pip install outlines_core --no-index --find-links dist --force-reinstall
+          python -c "import outlines_core"
+
+      - uses: actions/upload-artifact@v4
+        with:
+          path: dist/*.whl
+          name: wheels-linux-${{ matrix.platform.target }}
+
+  windows:
+    name: Build ${{ matrix.platform.runner}} ${{ matrix.platform.target }}
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      matrix:
+        platform:
+          - runner: windows-latest
+            target: x86
+            alias-target: i686-pc-windows-msvc
+            interpreter: "3.9 3.10 3.11 3.12 3.13"
+          - runner: windows-latest
+            target: x64
+            alias-target: x86_64-pc-windows-msvc
+            interpreter: "3.9 3.10 3.11 3.12 3.13"
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: "3.13"
+          architecture: ${{ matrix.platform.target }}
+
+      - run: pip install -U twine
+
+      - name: Install required packages
+        # rustls requires aws-lc-sys, which FFI bindings to AWS-LC (AWS Libcrypto)
+        # aws-lc-sys requires nasm for compilation on windows
+        run: |
+          choco install nasm
+
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: stable
+          target: ${{ matrix.platform.alias-target}}
+      - uses: Swatinem/rust-cache@v2
+
+      - name: Bump version in Cargo.toml
+        if: ${{ inputs.auto_bump }}
+        uses: ./.github/actions/bump_version
+
+      - name: Build wheels
+        uses: PyO3/maturin-action@v1
+        with:
+          target: ${{ matrix.platform.target }}
+          args: --release --out dist --interpreter ${{ matrix.platform.interpreter }}
+          sccache: ${{ !startsWith(github.ref, 'refs/tags/') }}
+          manylinux: auto
+
+      - name: Validate Python package distributions
+        run: twine check --strict dist/*
+
+      - name: Install built wheel
+        run: |
+          pip install outlines_core --no-index --find-links dist --force-reinstall
+          python -c "import outlines_core"
+
+      - uses: actions/upload-artifact@v4
+        with:
+          path: dist/*.whl
+          name: wheels-windows-${{ matrix.platform.target }}
+
+  macos:
+    name: Build ${{ matrix.platform.runner}} ${{ matrix.platform.target }}
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      matrix:
```

Comment:
Would have been nice to maybe actually write the following lines as a matrix of `[macOS 14, macOS 15] x [aarch64, x86_64]`?

Reply:
How do you mean?

Shuffle like this?
```
          - runner: macos-14
            target: x86_64
            macos_version: "14.0"
            interpreter: "3.9 3.10 3.11 3.12 3.13"
          - runner: macos-15
            target: x86_64
            macos_version: "15.0"
            interpreter: "3.9 3.10 3.11 3.12 3.13"
          - runner: macos-14
            target: aarch64
            macos_version: "14.0"
            interpreter: "3.9 3.10 3.11 3.12 3.13"
          - runner: macos-15
            target: aarch64
            macos_version: "15.0"
            interpreter: "3.9 3.10 3.11 3.12 3.13"
```

Reply:
More like?
```yaml
macos:
  name: Build ${{ matrix.macos_version }} ${{ matrix.target }}
  runs-on: ${{ matrix.runner }}
  strategy:
    matrix:
      macos_version: [14.0, 15.0]
      target: [x86_64, aarch64]
      interpreter: [3.9, 3.10, 3.11, 3.12, 3.13]
```

Reply:
It needs macos version label in a certain format, which should match macos-15 or macos-14 versions. Actually, using matrixes this way exactly introduced a bug mentioned before (that we'll have wheels for both macos versions this way).

From my perspective workflows should be as stupid as they can be, repetitions help to see clearly without extra math in the head, which is the main source of errors here.

And as a bonus, if we'd have to introduce one more key - it would be stupidly simple too.